src/Components/Endpoints/src/Builder/RazorComponentEndpointFactory.cs

```diff
@@ -55,16 +56,16 @@ internal class RazorComponentEndpointFactory
         // The display name is for debug purposes by endpoint routing.
         builder.DisplayName = $"{builder.RoutePattern.RawText} ({pageDefinition.DisplayName})";
 
-        builder.RequestDelegate = CreateRouteDelegate(pageDefinition.Type);
+        builder.RequestDelegate = CreateRouteDelegate(rootComponent, pageDefinition.Type);
```

Comment:
I think we should also add the root component to the endpoint metadata, probably with a new `RootComponentMetadata`.